lib/core/src/Cardano/Wallet.hs

```diff
@@ -2142,9 +2143,13 @@ mkTxMetaWithoutSel blockHeader wState txCtx inps outs =
                 w@WithdrawalSelf{} -> Coin.add (withdrawalToCoin w)
                 WithdrawalExternal{} -> Prelude.id
                 NoWithdrawal -> Prelude.id
+        dir = if (txCtx ^. #txDelegationAction) == Just Quit then
```

Comment:
This logic is called from `submitTransaction`, where the user could submit any kind of (wallet-related) transaction.

Defining the direction this way might work for txs from `constructTransaction`, but with `balanceTransaction` it would quickly break.

What was the reason we can't use `mkTxMeta` instead of `constructTxMeta`? (which itself is duplicated with logic in Primitive.Model)

Reply:
well, we are not doing selection when submitting, we are decoding what is inside tx cbor to be sent. hence we do not use, `mkTxMeta`. But, we can resign from `dir` and compare `amtInps` and `amtOuts` to determine direction. Now it should survive `balanceTransaction` step (or at least we will not deteriorate situation in the context what was before this PR). I have refactored constructTxMeta and found a bug in submitTransaction. And simplified the things. And locally it is green